apiserver/facades/client/modelmanager/modelmanager.go

```diff
@@ -499,13 +506,24 @@ func (m *ModelManagerAPI) newCAASModel(cloudSpec environs.CloudSpec,
 		return nil, errors.Annotate(err, "failed to create config")
 	}
 
-	broker, err := caas.New(environs.OpenParams{
+	broker, err := m.getBroker(environs.OpenParams{
 		Cloud:  cloudSpec,
 		Config: newConfig,
 	})
 	if err != nil {
 		return nil, errors.Annotate(err, "failed to open kubernetes client")
 	}
+
+	// CAAS models exist in a namespace which must be unique.
+	namespaces, err := broker.Namespaces()
+	if err != nil {
+		return nil, errors.Annotate(err, "failed to list namespaces")
+	}
+	nsSet := set.NewStrings(namespaces...)
+	if nsSet.Contains(createArgs.Name) {
+		return nil, errors.NewAlreadyExists(nil, fmt.Sprintf("namespace called %q already exists, would clash with model name", createArgs.Name))
```

Comment:
Would not this message be stuttering "..already exists..already exists...."?

Reply:
No - the New method used is for the case where we want the error type to be AlreadyExists but we need a bespoke message. As opposed to AlreadyExistsf(). All of our error types support this.